src/main-process/atom-window.coffee

```diff
@@ -40,9 +40,15 @@ class AtomWindow
     if process.platform is 'linux'
       options.icon = @constructor.iconPath
 
+    if @applyTitleBarSetting()
+      options.titleBarStyle = 'hidden'
+
     @browserWindow = new BrowserWindow options
     global.atomApplication.addWindow(this)
 
+    if @applyTitleBarSetting()
+      @browserWindow.setSheetOffset(23)
```

Comment:
Fixed in https://github.com/atom/atom/pull/11790/commits/76357473ade7dadb6a2e318fb60ff48b704ece97